lib/mayaUsd/fileio/translators/translatorLight.cpp

```diff
@@ -0,0 +1,503 @@
+
+    UsdMayaWriteUtil::SetAttribute(
+            usdLight.GetRadiusAttr(), lightRadius, usdTime, valueWriter);
+    UsdMayaWriteUtil::SetAttribute(
+            usdLight.GetTreatAsPointAttr(), (lightRadius == 0.f), usdTime, valueWriter);
+    
+    UsdPrim prim = usdLight.GetPrim();
+    UsdLuxShapingAPI shapingAPI = UsdLuxShapingAPI::Apply(prim);
+
+    // We need some magic conversions between maya dropOff, coneAngle, penumbraAngle, 
+    // and Usd shapingFocus, shapingConeAngle, shapingConeSoftness
+    const double dropOff = mayaLight.dropOff(&status);
+    CHECK_MSTATUS_AND_RETURN(status, false);
+
+    if (dropOff > 0) {
+        UsdMayaWriteUtil::SetAttribute(
+            shapingAPI.CreateShapingFocusAttr(),
+            static_cast<float>(dropOff),
+            usdTime,
+            valueWriter);
+    }
+
+    double coneAngle = GfRadiansToDegrees(mayaLight.coneAngle(&status)) * 0.5;
+    CHECK_MSTATUS_AND_RETURN(status, false);
+    double penumbraAngle = GfRadiansToDegrees(mayaLight.penumbraAngle(&status));
+    CHECK_MSTATUS_AND_RETURN(status, false);
+
+    float cutoff = static_cast<float>(coneAngle + penumbraAngle);
+    UsdMayaWriteUtil::SetAttribute(
+        shapingAPI.CreateShapingConeAngleAttr(), cutoff, usdTime, valueWriter);
+
+    float softness = static_cast<float>(cutoff > 0 ? penumbraAngle / cutoff : 0.f);
+    if (softness > 0) {
+        UsdMayaWriteUtil::SetAttribute(
+            shapingAPI.CreateShapingConeSoftnessAttr(), softness, usdTime, valueWriter);
+    }
+
+    return true;
+}
+
+// Import the specialized MFnSpotLight attributes
+static bool _ReadSpotLight(const UsdLuxLight& lightSchema, 
+                            MFnDependencyNode &depFn,  
+                            const UsdMayaPrimReaderArgs& args,
+                            UsdMayaPrimReaderContext*    context)
+{
+
+    MStatus status;
+    bool success = true;
+    UsdPrim prim = lightSchema.GetPrim();
+    UsdLuxSphereLight sphereLight(prim);
+    if (!sphereLight) {
+        return false;
+    }
+    success &= UsdMayaReadUtil::ReadUsdAttribute(sphereLight.GetRadiusAttr(), depFn,
+                     _tokens->LightRadiusPlugName,
+                     args, context);
+
+    UsdLuxShapingAPI shapingAPI(prim);
+    if (!shapingAPI) {
+        return false;
+    }
+
+    // We need some magic conversions between maya dropOff, coneAngle, penumbraAngle, 
+    // and Usd shapingFocus, shapingConeAngle, shapingConeSoftness
+    success &= UsdMayaReadUtil::ReadUsdAttribute(shapingAPI.GetShapingFocusAttr(), depFn,
+                     _tokens->DropoffPlugName,
+                     args, context);
+
+    float UsdConeAngle = 1.f;
+    shapingAPI.GetShapingConeAngleAttr().Get(&UsdConeAngle);
+    float coneSoftness = 0.f;
+    shapingAPI.GetShapingConeSoftnessAttr().Get(&coneSoftness);
+
+    float penumbraAngle = UsdConeAngle * coneSoftness;
+    float mayaConeAngle = 2.f * (UsdConeAngle - penumbraAngle);
```

Comment:
A negative penumbra angle becomes positive on a roundtrip and this will adjust the cone angle accordingly, resulting in the exact same illumination. This is perfectly fine, but should probably be documented.

Reply:
I guess you were referring to online documentation, but I added a comment about this in the code